python/ray/_private/accelerator.py

```diff
@@ -194,31 +194,45 @@ def _autodetect_tpu_version() -> Optional[str]:
     """
 
     def accelerator_type_to_version(accelerator_type: str) -> str:
-        assert_tpu_accelerator_type(accelerator_type)
-        return "TPU-" + str(accelerator_type.split("-")[0]).upper()
+        if valid_tpu_accelerator_type(accelerator_type):
+            return "TPU-" + str(accelerator_type.split("-")[0]).upper()
+        else:
+            return None
 
+    detected_tpu_version = None
     # GKE-based check
     accelerator_type = os.getenv(
         ray_constants.RAY_GKE_TPU_ACCELERATOR_TYPE_ENV_VAR, None
     )
     if accelerator_type is not None:
-        return accelerator_type_to_version(accelerator_type)
-
-    # GCE-based VM check
-    try:
-        accelerator_type_request = requests.get(
-            ray_constants.RAY_GCE_TPU_ACCELERATOR_ENDPOINT,
-            headers=ray_constants.RAY_GCE_TPU_HEADERS,
-        )
-        if accelerator_type_request.status_code == 200:
-            return accelerator_type_to_version(accelerator_type_request.text)
-    except requests.RequestException as e:
-        logging.info("Unable to poll TPU GCE metadata: %s", e)
-
-    return None
-
-
-def assert_tpu_accelerator_type(accelerator_type: str):
+        detected_tpu_version = accelerator_type_to_version(accelerator_type)
+    else:
+        # GCE-based VM check
+        try:
+            accelerator_type_request = requests.get(
+                ray_constants.RAY_GCE_TPU_ACCELERATOR_ENDPOINT,
+                headers=ray_constants.RAY_GCE_TPU_HEADERS,
+            )
+            if (
+                accelerator_type_request.status_code == 200
+                and accelerator_type_request.text
+            ):
+                detected_tpu_version = accelerator_type_to_version(
+                    accelerator_type_request.text
+                )
+        except requests.RequestException as e:
+            logging.info(
+                "While trying to autodetect a TPU type, "
+                " unable to poll TPU GCE metadata: %s",
+                e,
+            )
+
+    if detected_tpu_version is None:
+        logging.info("Failed to auto-detect TPU type.")
```

Comment:
There are two ways for `detected_tpu_version` to be `None`:
1. `accelerator_type` was specified, but but it wasn't a valid accelerator type
2. `accelerator_type was not specified, but the GCE response didn't include a valid accelerator type

Do we need to distinguish these two cases in the info log?

Reply:
(This seems relatively unimportant and doesn't need to block the PR)

Reply:
Added in more logging anyways!